go/runtime/txpool/main_queue.go

```diff
@@ -114,6 +114,15 @@ func (mq *mainQueue) GetSchedulingExtra(offset *hash.Hash, limit uint32) []*TxQu
 	return txs
 }
 
+func (mq *mainQueue) PeekAll() []*TxQueueMeta {
+	allTxs := mq.inner.getAll()
+	var txs []*TxQueueMeta
```

Comment:
Preallocate the slice if you can get the size (also in `TakeAll`?).

Reply:
Done, I've also modified other `mainQueue` methods to preallocate their output slices.